### ISSUE TITLE
Episode data missing duration information

### ISSUE THREAD

**Issue by SarvagyaVaish** [Podcastindex-org/aggregator#5]
(not sure if this is the correct repo for this issue; if not, please let me know and I can repost in the appropriate repo.)

I have noticed a potential issue with the aggregator. I will use _How I Built This with Guy Raz_ as an example, but the issue might not be limited to this podcast. Most episodes of this podcast, when fetched using the episodes api have the duration information reflected correctly. Some episodes do not.

Below is all the information needed to reproduce the issue. Do you know what the underlying issue is?

Podcast Itunes ID: 1150510297
Episode ID: 862160048
API called: https://api.podcastindex.org/api/1.0/episodes/byid?id=862160048
Response (see "duration" key has value "0"): 
```
{
    "status": "true",
    "id": "862160048",
    "episode": {
        "id": 862160048,
        "title": "Remembering Tony Hsieh of Zappos",
        "link": "https://www.npr.org/2020/11/24/938418035/remembering-tony-hsieh-of-zappos",
        "description": "The former CEO of Zappos ...",
        "guid": "3f81470b-a323-4537-9115-3d2c805fe495",
        "datePublished": 1606712468,
        "datePublishedPretty": "November 29, 2020 11:01pm",
        "dateCrawled": 1606720479,
        "enclosureUrl": "https://play.podtrac.com/npr-510313/edge1.pod.npr.org/anon.npr-podcasts/podcast/npr/hibt/2020/11/20201129_hibt_hibt_-_zappos__ramtin___-_final_podcast_mix_2020_repeat__rip_-a93b8f59-36c2-4a98-9f23-1bc0a13a97aa.mp3?awCollectionId=510313&amp;awEpisodeId=938418035&amp;orgId=1&amp;d=1769&amp;p=510313&amp;story=938418035&amp;t=podcast&amp;e=938418035&amp;size=28252341&amp;ft=pod&amp;f=510313",
        "enclosureType": "audio/mpeg",
        "enclosureLength": 28252341,
        "duration": 0,
        "explicit": 0,
        "episode": null,
        "episodeType": "full",
        "season": 0,
        "image": "https://media.npr.org/assets/img/2020/11/29/ep20-zappos_wide-521dbc884cb0d45683f515513d2a99528f33acce.jpeg?s=1400",
        "feedItunesId": 1150510297,
        "feedImage": "https://media.npr.org/assets/img/2018/08/03/npr_hibt_podcasttile_sq-98320b282169a8cea04a406530e6e7b957665b3f.jpg?s=1400",
        "feedId": 714258,
        "feedTitle": "How I Built This with Guy Raz",
        "feedLanguage": "en-us",
        "chaptersUrl": null,
        "transcriptUrl": null
    },
    "description": "Found matching item."
}
```

FeedUrl: https://feeds.npr.org/510313/podcast.xml
Screenshot of rss item.

![image](https://user-images.githubusercontent.com/1449905/101299134-e2615480-37e5-11eb-84dd-1e7b95925299.png)

**Comment by daveajones** [comment]
Weird.  Let me investigate this.  I did some testing and the durations seem to come through correctly when given those inputs.  I'll dig deeper and figure out what's going on.

**Comment by daveajones** [comment]
And thank you for the very detailed bug report.  So helpful!

**Comment by daveajones** [comment]
Ok, I think I found the issue.  It was a type checking bug.  I forced a feed check after the fix and a new episode is up in that feed and it picked up the duration correctly this time.  I don't think the itunes spec officially allows for using total seconds in this tag.  It wants you to use "HH:MM:SS" format as a string.  But, I see it done this way often enough that this should be considered a bug.

Thanks!

**Comment by SarvagyaVaish** [comment]
Awesome! Thanks for investigating. 
What should I expect next? When will the API reflect the change? I naively checked this url, and the durations is still 0.
https://api.podcastindex.org/api/1.0/episodes/byid?id=862160048

**Comment by daveajones** [comment]
I’ll have to script that change. Normally old episodes don’t update unless the guid changes.  Give me a day or so to get that done if you can.

**Comment by survyvaish** [comment]
Gotcha. Thanks! 
Happy to help code it up if you point me in the right direction. I love this project and am planning to use it extensively for a project I have in mind.

**Comment by survyvaish** [comment]
Could there be a write endpoint I can call with a list of episodes to reindex if I notice the duration field is off?

**Comment by SarvagyaVaish** [comment]
(Oops commented from a different account earlier)
Gotcha. Thanks!
Happy to help code it up if you point me in the right direction. I love this project and am planning to use it extensively for a project I have in mind.
Could there be a write endpoint I can call with a list of episodes to reindex if I notice the duration field is off?

**Comment by daveajones** [comment]
There is a notify endpoint to tell the index to rescan a feed, but it doesn't override what's already there.  I'm actually working on adding a parameter to it for that as we speak.  I'll ping you when it's ready.  It'll let you tell the index to re-scan and overwrite.

**Comment by daveajones** [comment]
BTW, we discuss a lot of these things at podcastindex.social, our mastodon instance.  It's the official community discussion for the project if you want to join there.

**Comment by SarvagyaVaish** [comment]
> BTW, we discuss a lot of these things at podcastindex.social, our mastodon instance. It's the official community discussion for the project if you want to join there.

I signed up but was a bit confused as to how it works. Can you tag me in a post?
https://podcastindex.social/web/accounts/8071

**Comment by comster** [comment]
For the record, I find mastodon a lot more difficult, hard to follow, and
has many conversations lost, compared to issue threads in github.

**Comment by daveajones** [comment]
Not going to argue that.  You're right.  Just faster when it comes to a quick back and forth.

**Comment by daveajones** [comment]
You can now tell the index to update existing feed items by passing the "&update" param to the websub notification point, like so:

https://api.podcastindex.org/api/1.0/hub/pubnotify?id=714258&update

The default is to pull/parse the feed and only look for new episodes.

If you pass the "update" param (no value needed) it will also refresh all** the existing episode metadata.

** Not actually all. Transcripts and chapters aren't updated yet.  Will get that next.
